lib/http.c

```diff
@@ -1596,6 +1596,8 @@ CURLcode Curl_http_done(struct Curl_easy *data,
        read from the HTTP server (that counts), this can't be right so we
        return an error here */
     failf(data, "Empty reply from server");
+    /* Mark it as closed to avoid the "left intact" message */
+    connclose(conn, "Empty reply from server");
```

Comment:
At this point is it possible that an http2 stream may be bad but the connection still good?

Reply:
Hm, fair point. I *think* we could in theory get it even with h2 and h3 so I'll change the call to streamclose... Thanks!

Reply:
AFAICS streamclose calls [Curl_conncontrol](https://github.com/curl/curl/blob/curl-7_74_0/lib/connect.c#L1578) which doesn't actually signal to close the stream (if I'm reading it right), so I don't think it's going to stop that message

Reply:
the message `connection left intact` is still there and meant for the actual **connection**, so a stream getting closed early or not will not necessarily affect the connection the stream was sent over. So I think this is fine...